### DISCUSSION TITLE
Rust error diagnostics aren't showing - any recommendations on how to debug?

### DISCUSSION THREAD

**Discussion by spencerkohan** [zed-industries/zed#7051] [Feature Requests]
I'm not sure exactly when it stopped working, sometime ago my rust error diagnostics stopped displaying correctly in the editor.

The strange thing is it's not that rust-analyzer isn't working at all, or that I have zero diagnostics, but I'm just not getting the ones I want.  I.e. I have some errors reported which are not actually errors when building from the comand line, but I don't see the actual errors in my code while I am typing.

It's worth noting that I do see the correct errors in VSCode.

The only thing which I could think might be an issue is that I'm on nightly - not sure if there is any additional configuration required.

Here's my settings.json - I'm not doing anything exotic:

```
{
  "buffer_font_size": 14,
  "soft_wrap": "editor_width",
  "language_overrides": {
    "TypeScript": {
      "enable_language_server": false
    },
    "TSX": {
      "enable_language_server": false
    },
    "JavaScript": {
      "enable_language_server": false
    }
  }
}
```

Has anyone by chance run into this and might be able to recommend a way forward?

**Comment by SomeoneToIgnore** [comment]
A good start would be Zed logs and LSP logs (both available via the command palette menu via cmd-p).

**Comment by alexpyattaev** [comment]
There is literally nothing "suspicious" in the logs. Another interesting is that the recommendations and type insets are not going anywhere, so clearly LSP is running. Seems like interaction with cargo check is flaky.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
